source/Patches/EndGameSummary.cs

```diff
@@ -0,0 +1,141 @@
+        }
+
+        public static void UpdatePlayerInfo()
+        {
+            Il2CppSystem.Collections.Generic.List<GameData.PlayerInfo> players = GameData.Instance.AllPlayers;
+
+            var pinfolist = new List<AdditionalTempData.PlayerInfo>();
+
+            foreach (GameData.PlayerInfo player in players)
+            {
+                byte pid = player.PlayerId;
+                string name = player.PlayerName;
+                Role role = Role.GetRole(player);
+                Modifier modifier = Modifier.GetModifier(player);
+                Faction faction = role.Faction;
+                int tasksdone = 0;
+                int taskstotal = 0;
+
+                AdditionalTempData.PlayerInfo pinfo = new AdditionalTempData.PlayerInfo();
+                pinfo.PlayerName = name;
+                pinfo.CurrentRole = role;
+                pinfo.Faction = faction;
+                pinfo.Modifier = modifier;
+
+                foreach (var task in player.Tasks)
+                {
+                        if (task.Complete == true)
+                        {
+                            tasksdone++;
+                        }
+                        taskstotal++;
+                }
+                
+                pinfo.TasksCompleted = tasksdone;
+                pinfo.TasksTotal = taskstotal;
+                pinfolist.Add(pinfo);
+            }
+            AdditionalTempData.playerData = pinfolist;
+        }
+
+        public static bool HasTasks(Faction faction)
+        {
+            return faction == Faction.Crewmates;
+        }
+
+        static bool IsNull(AdditionalTempData.PlayerInfo pinfo, string type)
+        {
+            return type switch {
+                "Role" => pinfo.CurrentRole == null,
+                "Mod" => pinfo.Modifier == null,
+                _ => true
+            };
+        }
+
+        public static void LoadGameSummary(EndGameManager __instance)
+        {
+            var position = Camera.main.ViewportToWorldPoint(new Vector3(0f, 1f, Camera.main.nearClipPlane));
+            GameObject roleSummary = UnityEngine.Object.Instantiate(__instance.WinText.gameObject);
+            roleSummary.transform.position = new Vector3(__instance.ExitButton.transform.position.x + 0.1f, position.y - 0.1f, -14f);
+            roleSummary.transform.localScale = new Vector3(1f, 1f, 1f);
+
+            var roleSummaryText = new StringBuilder();
+
+            roleSummaryText.AppendLine("End-of-game Summary:");
+
+            foreach (var data in AdditionalTempData.playerData)
+            {
+                var won = "   ";
+                foreach (var winner in TempData.winners)
+                {
+                    if (winner.Name == data.PlayerName)
+                    {
+                        won = $"[<color=#FAD934FF>W</color>]";
+                    }
+                }
+                var nameinfo = data.PlayerName;
+
+                List<string> noncrewroleswithtasks = new List<string>();
```

Comment:
Nit: Consider camelCase for variable names to make them easier to read.

Reply:
usually do, forgot it here, will edit when i get a chance.